tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -115,23 +116,35 @@ initialize_column_state(DecompressChunkState *state)
 	ScanState *ss = (ScanState *) state;
 	TupleDesc desc = ss->ss_ScanTupleSlot->tts_tupleDescriptor;
 	ListCell *lc;
-	int i;
 
-	state->num_columns = list_length(state->varattno_map);
+	if (list_length(state->decompression_map) == 0)
+	{
+		elog(ERROR, "no columns specified to decompress");
```

Comment:
This is an internal program logic error, like an Assert. It shouldn't be hit by tests.

Reply:
We have more of those. Not sure if we can mark them so that they do not show up on codecov. Might make our work easier.

Reply:
It would also make fuzzing easier -- if you hit a logic error when fuzzing, it's definitely a bug, but it's OK to hit user errors. IIRC postgres doesn't have the proper separation between user or environment errors, and program logic errors. The convention is to use `elog(ERROR, ...)` for the latter, but I'm not sure they have a corresponding error code set that would allow to distinguish them programmatically.

Reply:
I think at timescale we often ignore this convention and use `elog(ERROR` for everything.

Reply:
If you want to trace hitting logic bugs while fuzzying, that would mean adding an annotation to the error messages. I suggested a macro for this purpose in https://github.com/timescale/timescaledb/pull/2018. Maybe we should pick this up again.

Reply:
I just checked and at least PG14 uses `ERRCODE_INTERNAL_ERROR` for plain `elog(ERROR`, so they should be distinguishable in postgres convention. We could start following it as well, although reviewing all the existing errors could be difficult.

Reply:
I think that sounds like a good idea. No need to change other errors in this PR.